src/ecdsa/keys.py

```diff
@@ -870,30 +873,36 @@ def from_pem(cls, string, hashfunc=sha1):
         :return: Initialised SigningKey object
         :rtype: SigningKey
         """
-        # the privkey pem may have multiple sections, commonly it also has
-        # "EC PARAMETERS", we need just "EC PRIVATE KEY".
         if not PY2 and isinstance(string, str):
             string = string.encode()
-        privkey_pem = string[
-            string.index(b("-----BEGIN EC PRIVATE KEY-----")) :
-        ]
-        return cls.from_der(der.unpem(privkey_pem), hashfunc)
+
+        # the privkey pem may have multiple sections, commonly it also has
+        # "EC PARAMETERS", we need just "EC PRIVATE KEY".
+        ec_private_key_index = string.find(b"-----BEGIN EC PRIVATE KEY-----")
+        if ec_private_key_index != -1:
+            return cls.from_der(
+                der.unpem(string[ec_private_key_index:]), hashfunc, pkcs8=False
+            )
+
+        private_key_index = string.find(b"-----BEGIN PRIVATE KEY-----")
+        if private_key_index != -1:
+            return cls.from_der(
+                der.unpem(string[private_key_index:]), hashfunc, pkcs8=True
+            )
+
+        raise ValueError("No EC PRIVATE KEY or PRIVATE KEY section in PEM")
 
     @classmethod
-    def from_der(cls, string, hashfunc=sha1):
+    def from_der(cls, string, hashfunc=sha1, pkcs8=False):
         """
         Initialise from key stored in :term:`DER` format.
 
-        Note, the only DER format supported is the RFC5915
-        (the sslay format) supported by OpenSSL, the more common PKCS#8 format
-        is NOT supported (see:
-        https://github.com/warner/python-ecdsa/issues/113 )
+        The DER formats supported are the un-encrypted RFC5915
+        (the sslay format) supported by OpenSSL, and the more common RFC5958
+        (the PKCS #8 format).
```

Comment:
the PKCS#8 format also supports encryption, something this code does not, the documentation should reflect that

Reply:
Will do